compiler/optimizer/ValuePropagationCommon.cpp

```diff
@@ -136,18 +136,17 @@ TR::ValuePropagation::ValuePropagation(TR::OptimizationManager *manager)
      _arrayCloneCalls(trMemory()),
      _objectCloneTypes(trMemory()),
      _arrayCloneTypes(trMemory()),
-     _bcdSignConstraints(NULL),
      _parmInfo(NULL),
      _parmTypeValid(NULL),
      _constNodeInfo(comp()->allocator())
    {
-   _vcHandler.setVP(this);
+   _vcHandler.setVP((TR::ValuePropagation *)this);
```

Comment:
We need to pass in the right instance of ValuePropagation to setVP to ensure correct behavior for its virtual methods for a given project. We can either add the following to the constructor of the derived class of ValuePropagation or its initialize() method:

_vcHandler.setVP(this);